m4/pbs_mach_type.m4

```diff
@@ -43,10 +43,7 @@ AC_DEFUN([PBS_AC_MACHINE_TYPE],
   AC_MSG_RESULT([$PBS_MACH])
   AC_SUBST(PBS_MACH)
   AC_DEFINE_UNQUOTED([PBS_MACH], ["$PBS_MACH"], [PBS machine type])
-  AS_CASE([$PBS_MACH],
-    [aix*],
-      [mom_mach_libs="-lodm -lcfg"],
-    [mom_mach_libs=""])
+    [mom_mach_libs=""]
```

Comment:
I don't think we need this this file and buildutils/pbs_mach_type anymore as no one is using `@PBS_MACH@` or `@mom_mach_libs@` for substitute except one place in resmom/Makefile.am which can be removed.

Reply:
We might choose to remove darwin later, but irrespective of that, i think we want to keep this file to keep a placeholder (atleast that) of where platform/mach type will be detected.